### DISCUSSION TITLE
Some custom utilities not working

### DISCUSSION THREAD

**Discussion by Say-M** [tailwindlabs/tailwindcss#11284] [Help]
Node: v18.15.0
Next.js: 13.4.0
tailwindcss: v3.3.2
postcss: 8.4.23
Browser: Chrome, Firefox
OS: Windows

I created some custom utilities using tailwind `@layer`
but some of them are not compiling.
For example:
I have `.btn` utility it compiling properly
but when I define `.btn-sm/.btn-lg` utilities then its not compiling.
but instead of  `.btn-sm/.btn-lg` class if I write `.sm/.lg` then its compiling properly.

here is using `.btn-sm` class its not compiling
![image](https://github.com/tailwindlabs/tailwindcss/assets/63895803/ce0a60cd-ce7e-4a03-b2c0-9c0a122de5ad)
![image](https://github.com/tailwindlabs/tailwindcss/assets/63895803/7dfe1c36-868a-42a1-909a-9f81f471f390)

here is using only `.sm` class its compiling
![image](https://github.com/tailwindlabs/tailwindcss/assets/63895803/10a195da-e0fa-43ce-b0df-0ea973138f7b)
![image](https://github.com/tailwindlabs/tailwindcss/assets/63895803/ac84499b-76a7-47d4-bebd-56937536db4c)

**Comment by adamwathan** [comment]
Hey! You're going to have to provide a proper reproduction as it seems to work fine in my testing:

https://play.tailwindcss.com/wpuOSLzn3H?file=css

<img width="974" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/4323180/e342c93d-c3bf-4603-b402-df855cdcf1da">
